src/wui/mapview.cc

```diff
@@ -587,51 +587,46 @@ Widelands::NodeAndTriangle<> MapView::track_sel(const Vector2i& p) {
 }
 
 bool MapView::scroll_map() {
-	// If one of the arrow keys is pressed, scroll this distance
-	uint32_t ScrollDistanceY = g_gr->get_yres() / 4;
-	uint32_t ScrollDistanceX = g_gr->get_xres() / 4;
-
-	if ((get_key_state(SDL_SCANCODE_UP) && get_key_state(SDL_SCANCODE_LEFT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_7) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(-ScrollDistanceX, -ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if ((get_key_state(SDL_SCANCODE_UP) && get_key_state(SDL_SCANCODE_RIGHT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_9) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(ScrollDistanceX, -ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if ((get_key_state(SDL_SCANCODE_DOWN) && get_key_state(SDL_SCANCODE_LEFT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_1) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(-ScrollDistanceX, ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if ((get_key_state(SDL_SCANCODE_DOWN) && get_key_state(SDL_SCANCODE_RIGHT)) ||
-	    (get_key_state(SDL_SCANCODE_KP_3) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(ScrollDistanceX, ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_UP) ||
-	    (get_key_state(SDL_SCANCODE_KP_8) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(0, -ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_DOWN) ||
-	    (get_key_state(SDL_SCANCODE_KP_2) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(0, ScrollDistanceY), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_LEFT) ||
-	    (get_key_state(SDL_SCANCODE_KP_4) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(-ScrollDistanceX, 0), Transition::Smooth);
-		return true;
-	}
-	if (get_key_state(SDL_SCANCODE_RIGHT) ||
-	    (get_key_state(SDL_SCANCODE_KP_6) && !(SDL_GetModState() & KMOD_NUM))) {
-		pan_by(Vector2i(ScrollDistanceX, 0), Transition::Smooth);
-		return true;
-	}
-	return false;
+	// arrow keys
+	const bool kUP = get_key_state(SDL_SCANCODE_UP);
+	const bool kDOWN = get_key_state(SDL_SCANCODE_DOWN);
+	const bool kLEFT = get_key_state(SDL_SCANCODE_LEFT);
+	const bool kRIGHT = get_key_state(SDL_SCANCODE_RIGHT);
+
+	// numpad keys
+	const bool kNumlockOff = !(SDL_GetModState() & KMOD_NUM);
+	const bool kNP1 = get_key_state(SDL_SCANCODE_KP_1);
```

Comment:
Just one more suggestion (feel free to ignore):
```
#define kNP(x) const bool kNP##x = kNumlockOff && get_key_state(SDL_SCANCODE_KP_##x);
kNP(1) kNP(2) … kNP(9)
```
This saves some repetitive code and you can get rid of the many `kNumlockOff` checks in the `if`s below

Reply:
Thanks, for this suggestion.